File: 02_arrays/arrays_solutions/print-all-subarrays/PrintAllSubarrays.java
/*
 * Time complexity: O(N^2)
 * Space complexity: O(1)
 */

public class PrintAllSubarrays {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4 };

        for (int i = 0; i < arr.length; i++) {
            StringBuilder str = new StringBuilder();
            for (int j = i; j < arr.length; j++) {
                str = str.append(arr[j]).append(" ");
                System.out.println(str);
            }
        }
    }
}

File: 02_arrays/arrays_solutions/print-all-subarrays/PrintAllSubArrays.js
function printAllSubArrays(arr) {
	for (let i = 0; i < arr.length; i++) {
		let str = "";
		for (let j = i; j < arr.length; j++) {
			str += arr[j] + " ";
			console.log(str);
		}
	}
}

printAllSubArrays([1, 2, 3, 4]);
